databroker/broker.py

```diff
@@ -501,6 +501,28 @@ def process(self, headers, func, fields=None, fill=False):
         _process(mds=self.mds, fs=self.fs, headers=headers, func=func,
                  fields=fields, fill=fill)
 
+    def export(headers, mds, fs):
+        """ export a list of headers
+
+            Parameters:
+            -----------
+            headers : databroker.header
+                list of headers that are going to be exported
+            mds : metadatasote
+                metadata store object
+            fs : file store
+                file store object
+        """
+        for header in headers:
+            mds.insert_run_start(**header['start'])
+            for descripto in header['descriptor']:
+                mds.insert_descriptor(**descriptor)
+                events = self.get_events(header)
+                for event in events:
+                    mds.insert_event(event, descriptor)
```

Comment:
The `insert_event` API works the same as the other insert methods: it takes `**event`. The `descriptor` arg in its signature refers to `event['descriptor']`, which is the uid for `descriptor`.

Also, this for loop should be un-indented. `get_events` returns events for *all* descriptors, in bulk . It does not need to be repeated for each descriptor.